src/Umbraco.Web.UI.Client/src/common/services/focus.service.js

```diff
@@ -0,0 +1,72 @@
+/**
+ @ngdoc service
+ * @name umbraco.services.focusService
+ *
+ * @description
+ * <b>Added in Umbraco 8.1</b>. Application-wide service for focus related stuff.
+ * 
+ */
+
+(function () {
+    "use strict";
+
+    function focusService() {
+        
+        
+        function focusInApp(e) {
+            service.currentFocus = e.target;
+        }
+        document.addEventListener('focusin', focusInApp);
```

Comment:
Should this use the angular $document (https://docs.angularjs.org/api/ng/service/$document) that is testable instead of the native DOM `document` ? (would love to start having more unit tests written for the js stuff, services are the easiest to test)

Reply:
The reason is because the `$document` object is testable but the native `document` object is not. unit testing requires mocking things which is why angular has $document and $window which wrap the underlying native objects but also allow you to mock methods on them so we can test. Generally speaking in angular you should never reference `document` or `window` directly (sometimes that is not possible but in most cases it is)

Reply:
but yes, fine for now